lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -118,6 +118,29 @@ void ConditionalSESERegion::print(llvm::raw_ostream &OS, unsigned indent) const
 // CFG Canonicalization Implementation
 //===----------------------------------------------------------------------===//
 
+// Our partitioning and other transformations can leave around lots of
+// unconditional branches between blocks that formerly had control edges.  Go
+// through and merge those to make later passes simpler.
+bool tf::contractUncondBranches(SILFunction *fn, DominanceInfo* DI, SILLoopInfo *LI) {
+  bool changed = false;
+  // Iterate carefully to avoid invalidating iterators: we mutate the block list
+  // while we walk it.
+  for (auto bbi = fn->begin(), e = fn->end(); bbi != e;) {
+    auto *bb = &*bbi;
+    if (mergeBasicBlockWithSuccessor(bb, DI, LI)) {
```

Comment:
the old code does not use mergeBasicBlockWithSuccessor() -- it may be worth mentioning this "less obvious refactoring" in PR description.

From a quick read of that func impl, it seems to be equivalent to the old code.

Reply:
Updated the description.